plugins/Web/plugin.py

```diff
@@ -149,38 +149,50 @@ def noIgnore(self, irc, msg):
     def getTitle(self, irc, url, raiseErrors):
         size = conf.supybot.protocols.http.peekSize()
         timeout = self.registryValue('timeout')
-        (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
-                timeout=timeout)
-        try:
-            text = text.decode(utils.web.getEncoding(text) or 'utf8',
-                    'replace')
-        except UnicodeDecodeError:
-            pass
-        if minisix.PY3 and isinstance(text, bytes):
-            if raiseErrors:
-                irc.error(_('Could not guess the page\'s encoding. (Try '
-                        'installing python-charade.)'), Raise=True)
-            else:
-                return None
         try:
-            parser = Title()
-            parser.feed(text)
-        except UnicodeDecodeError:
-            # Workaround for Python 2
-            # https://github.com/ProgVal/Limnoria/issues/1359
-            parser = Title()
-            parser.feed(text.encode('utf8'))
-        parser.close()
-        title = utils.str.normalizeWhitespace(''.join(parser.data).strip())
-        if title:
-            return (target, title)
-        elif raiseErrors:
-            if len(text) < size:
-                irc.error(_('That URL appears to have no HTML title.'),
-                        Raise=True)
-            else:
-                irc.error(format(_('That URL appears to have no HTML title '
-                                 'within the first %S.'), size), Raise=True)
+            (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
+                timeout=timeout)
+        except Exception as e:
+            self.log.info('Web plugin TitleSnarfer: URL <%s> raised <%s>', url, str(e))
+            text = None
+        if text:
+            try:
+                text = text.decode(utils.web.getEncoding(text) or 'utf8',
+                        'replace')
+            except UnicodeDecodeError:
+                pass
+            if minisix.PY3 and isinstance(text, bytes):
```

Comment:
this condition should be in the `except` clause (and so you can drop the `isinstance`).

Reply:
Git doesn't helped in showing the changes I have made so just to make it clear: I didn't changed this part in this pull request, it's the same as before. Of course I can try changing it from now, just tell me if it's better if I change it as part of this pull request, create another after this one or if I leave it alone.

Reply:
Oh ok. That's up to you then

Reply:
Changed accordingly your suggestion.